contractsdk/cpp/XRC/xrc01/src/main/xrc01_e1.cc

```diff
@@ -0,0 +1,219 @@
+     } else if (var == "false" || var == "FALSE") {
+         var_bool = false;
+         return true;
+     }
+     return false;
+}
+
+struct XRC01_E1 : public xchain::Contract {};
+
+DEFINE_METHOD(XRC01_E1, initialize) {
+    xchain::Context* ctx = self.context();
+    const std::string& creator = ctx->arg("creator");
+    if (creator.empty()) {
+        ctx->error("missing creator");
+        return;
+    }
+    ctx->put_object("creator", creator);
+    ctx->ok("initialize succeed");
+}
+
+DEFINE_METHOD(XRC01_E1, issue) {
+    xchain::Context* ctx = self.context();
+    XRC01 xrc01(ctx);
+    XRC01::XRC01::token token;
+
+    const std::string& fungible_str = ctx->arg("fungible");
+    bool fungible;
+    if (!str2bool(fungible_str, fungible)) {
+        printf("issue token failed, fungible params illegal!");
+        ctx->error("issue token failed fungible params illegal!");
+        return;
+    }
+
+    uint64_t id= std::stoull(ctx->arg("id"));
+    const std::string& name = ctx->arg("name");
+    uint64_t supply = std::stoull(ctx->arg("supply"));
+    const std::string& issue_account = ctx->arg("issue_account");
+    const std::string& profile_desc = ctx->arg("profile_desc");
+    if (name.empty() || issue_account.empty() || profile_desc.empty()) {
+        ctx->error("issue error, param error");
+        return;
+    }
+
+    token.set_id(id);
+    token.set_name(name);
+    token.set_fungible(fungible);
+    token.set_supply(supply);
+    token.set_issue_account(issue_account);
+    token.set_profile_desc(profile_desc);
+
+    if (!xrc01.issue(token)) {
+        printf("issue token failed!");
+        ctx->error("issue token failed!");
+        return;
+    }
+    ctx->ok("issue succeed");   
+}
+
+DEFINE_METHOD(XRC01_E1, authorization) {
+    xchain::Context* ctx = self.context();
+    XRC01 xrc01(ctx);
+    const std::string& to = ctx->arg("to");
+    uint64_t token_id = std::stoll(ctx->arg("token_id"));
+    uint64_t amount = std::stoll(ctx->arg("amount"));
+
+    if (!xrc01.authorization(to, token_id, amount)) {
+        printf("authorization token failed!");
+        ctx->error("authorization token failed!");
+        return;
+    }    
+    ctx->ok("authorization succeed");   
+}
+
+DEFINE_METHOD(XRC01_E1, withdraw_authorization) {
+    xchain::Context* ctx = self.context();
+    XRC01 xrc01(ctx);
+
+    const std::string& from = ctx->arg("from");
+    uint64_t token_id = std::stoll(ctx->arg("token_id"));
+    uint64_t amount = std::stoll(ctx->arg("amount"));
+
+    if (!xrc01.withdraw_authorization(from, token_id, amount)) {
+        printf("withdraw_authorization token failed!");
+        ctx->error("withdraw_authorization token failed!");
+        return;
+    }    
+    ctx->ok("withdraw_authorization succeed");  
+}
+
+DEFINE_METHOD(XRC01_E1, transfer) {
+    xchain::Context* ctx = self.context();
+    XRC01 xrc01(ctx);
+
+    const std::string& to = ctx->arg("to");
+    uint64_t token_id = std::stoll(ctx->arg("token_id"));
```

Comment:
check overflow. 
"strtoll" is safer, but not enough though.  refer to the wiki I shared in the hi group.

Reply:
ok, I will fix it by providing a safe `stoull` in `safemath` package.